var/spack/repos/builtin/packages/hic-pro/package.py

```diff
@@ -0,0 +1,83 @@
+# it under the terms of the GNU Lesser General Public License (as
+# published by the Free Software Foundation) version 2.1, February 1999.
+#
+# This program is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the IMPLIED WARRANTY OF
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the terms and
+# conditions of the GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with this program; if not, write to the Free Software
+# Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
+##############################################################################
+from spack import *
+
+
+class HicPro(MakefilePackage):
+    """HiC-Pro is a package designed to process Hi-C data,
+    from raw fastq files (paired-end Illumina data)
+    to the normalized contact maps"""
+
+    homepage = "https://github.com/nservant/HiC-Pro"
+    url      = "https://github.com/nservant/HiC-Pro/archive/v2.10.0.tar.gz"
+
+    version('2.10.0', '6ae2213dcc984b722d1a1f65fcbb21a2')
+
+    depends_on('bowtie2')
+    depends_on('samtools')
+    depends_on('python@2.7:2.8')
+    depends_on('py-numpy')
+    depends_on('py-scipy')
+    depends_on('py-pysam')
+    depends_on('py-bx-python')
+    depends_on('r')
+    depends_on('r-rcolorbrewer')
+    depends_on('r-ggplot2')
+
+    def edit(self, spec, prefix):
+        makefile = FileFilter('Makefile')
+        makefile.filter('cp -Ri $(MK_PATH)',
+                        'cp -R $(MK_PATH)/*')
+        config = FileFilter('config-install.txt')
+        config.filter('PREFIX =.*', 'PREFIX = {0}'.format(prefix))
+        config.filter('BOWTIE2 PATH =.*',
+                      'BOWTIE2_PATH = {0}'.format(spec['bowtie2'].prefix))
+        config.filter('SAMTOOLS_PATH =.*',
+                      'SAMTOOLS_PATH = {0}'.format(spec['samtools'].prefix))
+        config.filter('R_PATH =.*',
+                      'R_RPTH ={0}'.format(spec['r'].prefix))
+        config.filter('PYTHON_PATH =.*',
+                      'PYTHON_RPTH ={0}'.format(spec['python'].prefix))
+
+    def build(self, spec, preifx):
+        make('-f', './scripts/install/Makefile',
+             'CONFIG_SYS=./config-install.txt')
+        make('mapbuilder')
+        make('readstrimming')
+        make('iced')
+
+    def install(sefl, spec, prefix):
+        # Patch INSTALLPATH in config-system.txt
+        config = FileFilter('config-system.txt')
+        config.filter('/HiC-Pro_2.10.0', '')
+        # Install
+        install('config-hicpro.txt', prefix)
+        install('config-install.txt', prefix)
+        install('config-system.txt', prefix)
+        install_tree('bin', prefix.bin)
+        install_tree('annotation', join_path(prefix, 'annotation'))
+        install_tree('doc', join_path(prefix, 'doc'))
+        install_tree('scripts', join_path(prefix, 'scripts'))
```

Comment:
You can actually use `prefix.annotation`, `prefix.doc`, and `prefix.scripts` if you want, no need for `join_path`. You'll still need it for `test-op` though.

Reply:
Fixed.